src/containers/TopbarContainer/Topbar/Topbar.js

```diff
@@ -244,6 +248,27 @@ const TopbarComponent = props => {
 
   const classes = classNames(rootClassName || css.root, className);
 
+  const { display: searchFormDisplay = 'always' } = config?.topbar?.searchBar || {};
+
+  const showSearchForm =
+    searchFormDisplay === SEARCH_DISPLAY_ALWAYS ||
+    (searchFormDisplay === SEARCH_DISPLAY_ONLY_SEARCH_PAGE &&
+      resolvedCurrentPage === 'SearchPage') ||
+    (searchFormDisplay === SEARCH_DISPLAY_NOT_LANDING_PAGE &&
+      resolvedCurrentPage !== 'LandingPage');
```

Comment:
It's difficult to understand what happens here.

Reply:
Added some comments to clarify the logic

Reply:
Rewrote the logic to use separate variables for each display config combination